config/webpack.dev.js

```diff
@@ -30,61 +31,66 @@ Object.assign(config, {
     libraryTarget: 'var',
     pathinfo: true,
   },
-});
-
-config.module.loaders.push({
-  test: /\.s[ac]ss$/,
-  use: [
-    'style-loader',
-    {
-      loader: 'css-loader',
-      options: {
-        minimize: false,
-        importLoaders: 1,
-        localIdentName: '[folder]__[local]___[hash:base64:5]',
-        modules: true,
+  module: {
+    loaders: [
+      {
+        test: /\.s[ac]ss$/,
+        use: [
+          'style-loader',
+          {
+            loader: 'css-loader',
+            options: {
+              minimize: false,
+              importLoaders: 1,
+              localIdentName: '[folder]__[local]___[hash:base64:5]',
+              modules: true,
+            },
+          },
+          {
+            loader: 'postcss-loader',
+            options: postCssOptions,
+          },
+          { loader: 'sass-loader' },
+        ],
+        include: [root('src')],
       },
-    },
-    {
-      loader: 'postcss-loader',
-      options: postCssOptions,
-    },
-    { loader: 'sass-loader' },
-  ],
-  include: [root('src')],
-});
-
-config.module.loaders.push({
-  test: /\.css$/,
-  use: [
-    'style-loader',
-    {
-      loader: 'css-loader',
-      options: {
-        minimize: false,
-        importLoaders: 1,
+      {
+        test: /\.css$/,
+        use: [
+          'style-loader',
+          {
+            loader: 'css-loader',
+            options: {
+              minimize: false,
+              importLoaders: 1,
+            },
+          },
+        ],
       },
-    },
+    ],
+  },
+  plugins: [
+    new webpack.HotModuleReplacementPlugin(),
+    new webpack.NoEmitOnErrorsPlugin(),
+    new webpack.NamedModulesPlugin(),
+    new webpack.WatchIgnorePlugin([
+      root('core'),
+      root('config'),
+      root('build'),
+    ]),
+    new webpack.DefinePlugin({
+      'process.env.DEV': true,
+      'process.env.BROWSER': true,
+      'process.env.NODE_ENV': JSON.stringify('development'),
+    }),
+    webpackIsomorphicToolsPlugin,
   ],
+}, (objValue, srcValue) => {
+  if (Array.isArray(objValue)) {
+    return objValue.concat(srcValue);
+  }
```

Comment:
better to extract a common function to do the customize, 
but I still think the old way is more readable